controller/monitor/snapshot_monitor.go

```diff
@@ -0,0 +1,588 @@
+package monitor
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/avast/retry-go"
+	"github.com/go-co-op/gocron"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/time/rate"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/client-go/util/workqueue"
+
+	etypes "github.com/longhorn/longhorn-engine/pkg/types"
+
+	"github.com/longhorn/longhorn-manager/constant"
+	"github.com/longhorn/longhorn-manager/datastore"
+	"github.com/longhorn/longhorn-manager/engineapi"
+	longhorn "github.com/longhorn/longhorn-manager/k8s/pkg/apis/longhorn/v1beta2"
+	"github.com/longhorn/longhorn-manager/types"
+	"github.com/longhorn/longhorn-manager/util"
+)
+
+type SnapshotHashStatus string
+
+const (
+	SnapshotHashStatusNew        = SnapshotHashStatus("")
+	SnapshotHashStatusInProgress = SnapshotHashStatus("in_progress")
+	SnapshotHashStatusCompleted  = SnapshotHashStatus("complete")
+	SnapshotHashStatusError      = SnapshotHashStatus("error")
+)
```

Comment:
NIT: Shouldn't these be in `longhorn-manager/k8s/pkg/apis/longhorn/v1beta2/snapshot.go` ?